table/index_builder.cc

```diff
@@ -101,7 +106,7 @@ void PartitionedIndexBuilder::AddIndexEntry(
     if (sub_index_builder_ != nullptr) {
       std::string handle_encoding;
       block_handle.EncodeTo(&handle_encoding);
-      bool do_flush =
+      bool do_flush = partition_cut_requested_ ||
```

Comment:
It is reset each time MakeNewSubIndexBuilder  is called, which is also called below after the flush.